src/archive_writer.rs

```diff
@@ -302,45 +351,217 @@ fn write_symbol_table_header<W: Write + Seek>(
         let pos = w.stream_position()?;
         print_bsd_member_header(w, pos, name, now(), 0, 0, 0, size)
     } else if is_aix_big_archive(kind) {
-        print_big_archive_member_header(w, "", now(), 0, 0, 0, size, prev_member_offset, 0)
+        print_big_archive_member_header(
+            w,
+            "",
+            now(),
+            0,
+            0,
+            0,
+            size,
+            prev_member_offset,
+            next_member_offset,
+        )
     } else {
         let name = if is_64bit_kind(kind) { "/SYM64" } else { "" };
         print_gnu_small_member_header(w, name.to_string(), now(), 0, 0, 0, size)
     }
 }
 
+fn compute_headers_size(
+    kind: ArchiveKind,
+    num_members: usize,
+    string_member_size: u64,
+    num_syms: u64,
+    sym_names_size: u64,
+    sym_map: Option<&SymMap>,
+) -> io::Result<u64> {
+    let offset_size = if is_64bit_kind(kind) { 8 } else { 4 };
+    let (symtab_size, _) =
+        compute_symbol_table_size_and_pad(kind, num_syms, offset_size, sym_names_size);
+    let compute_symbol_table_header_size = || -> io::Result<u64> {
+        let mut tmp = Cursor::new(Vec::new());
```

Comment:
You can `io::Write::write` directly to a `Vec`. No need for a `Cursor` here.

Reply:
Except `write_symbol_table_header` requires `Seek` (for `stream_position`), which `Vec` does not implement.